core/controllers/story_editor_test.py

```diff
@@ -46,6 +46,134 @@ def setUp(self):
 
 class StoryEditorTests(BaseStoryEditorControllerTests):
 
+    def test_can_not_access_story_editor_page_with_invalid_story_id(self):
+        self.login(self.ADMIN_EMAIL)
+
+        new_story_id = story_services.get_new_story_id()
+
+        self.get_html_response(
+            '%s/%s/%s' % (
+                feconf.STORY_EDITOR_URL_PREFIX, self.topic_id,
+                new_story_id), expected_status_int=404)
+
+        # Raises error 404 even when story is saved as the new story id is not
+        # associated with the topic.
+        self.save_new_story(
+            new_story_id, self.admin_id, 'Title', 'Description', 'Notes')
+
+        self.get_html_response(
+            '%s/%s/%s' % (
+                feconf.STORY_EDITOR_URL_PREFIX, self.topic_id,
+                new_story_id), expected_status_int=404)
+
+        self.logout()
+
+    def test_get_can_not_access_story_handler_with_invalid_story_id(self):
+        self.login(self.ADMIN_EMAIL)
+
+        new_story_id = story_services.get_new_story_id()
+
+        self.get_json(
+            '%s/%s/%s' % (
+                feconf.STORY_EDITOR_DATA_URL_PREFIX, self.topic_id,
+                new_story_id), expected_status_int=404)
+
+        # Raises error 404 even when story is saved as the new story id is not
+        # associated with the topic.
+        self.save_new_story(
+            new_story_id, self.admin_id, 'Title', 'Description', 'Notes')
+        self.get_json(
+            '%s/%s/%s' % (
+                feconf.STORY_EDITOR_DATA_URL_PREFIX, self.topic_id,
+                new_story_id), expected_status_int=404)
+
+        self.logout()
+
+    def test_put_can_not_access_story_handler_with_invalid_story_id(self):
```

Comment:
Same here.

In general, please prefer easy-to-read test names rather than names consistent with surrounding code. Tests are most helpful when they fail, so it's important they be as easy to read/comprehend in isolation.

Reply:
Done